packages/blade/src/components/Box/BaseBox/getBaseBoxStyles.tsx

```diff
@@ -0,0 +1,211 @@
+import type { CSSObject } from 'styled-components';
+import type {
+  BaseBoxProps,
+  MakeValueResponsive,
+  SpacingValueType,
+  ArrayOfMaxLength4,
+} from './types';
+import type { Breakpoints } from '~tokens/global/breakpoints';
+import breakpoints from '~tokens/global/breakpoints';
+import { getMediaQuery } from '~src/utils/getMediaQuery';
+import { isReactNative, isEmpty, getIn, makeSpace, makeSize } from '~utils';
+import type { Theme } from '~components/BladeProvider';
+
+const getResponsiveValue = <T extends string | number | string[]>(
+  value: MakeValueResponsive<T> | undefined,
+  size: keyof Breakpoints = 'base',
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): any | undefined => {
+  if (value === undefined || value === null) {
+    return undefined;
+  }
+
+  if (typeof value === 'string' || typeof value === 'number' || Array.isArray(value)) {
+    return value;
+  }
+
+  if (isEmpty(value)) {
+    return undefined;
+  }
+
+  if (isReactNative()) {
+    // In React Native, we map the value `s` token on priority (since the breakpoint maps to mobiles in useBreakpoint hook).
+    // We further look into smaller sizes, then we check base size, then we check medium, large and extra large sizes.
+    // Then we return the first non-undefined value in this priority
+    const priorityArray = [value.s, value.xs, value.base, value.m, value.l, value.xl];
```

Comment:
Not sure about:

> In React Native, it picks up props in this priority -> s > xs > base > m > l > xl

Can we have it similar to web? I suppose with the advent to devices we might have small, medium, large mobile devices so a consistent API similar to web perhaps. Or are there any tradeoffs you forsee with that 🤔

Reply:
Oh good point. Actually we should change the logic here to match web.

Imagine this. Expected behaviour would be that value will be undefined till m and then it will be none.
```jsx
<Box display={{ m: 'none' }}>
```

But on react native, by current logic, it will read none and set none which is unexpected. 

I feel logic on react native would be - `value.s ?? value.xs ?? value.base ?? undefined`

Reply:
Didn't get this part:

> I feel logic on react native would be - value.s ?? value.xs ?? value.base ?? undefined

> Actually we should change the logic here to match web.

Yup, would a 1:1 mapping not work for some cases?

Reply:
changed it to this - `value.s ?? value.xs ?? value.base ?? undefined`

Reply:
> Yup, would a 1:1 mapping not work for some cases?

So this logic is there because the media queries don't work on React Native. We return nothing from media query function in React Native.

So we need a separate logic to handle responsiveness in React Native. Doing `value.s ?? value.xs ?? value.base ?? undefined` means, we first look in `s` breakpoint then `xs -> base -> undefined`. This will be similar to how mobile styling works in mweb. We look into last matching min-width (last breakpoint in mobiles is `s`), then go looking down.

Reply:
I see, the conventional media queries won't work on Native. We may need different responsive approach probably using [`Dimensions`](https://reactnative.dev/docs/dimensions).

Reply:
We can do this in a followup PR, not crucial for this PR

Reply:
yup. Not sure if we would need `Dimensions` because the screen sizes would most probably stay similar in React Native unless we start using React Native desktop or web. There might be certain cases where it would be needed for small mobiles but I guess we can handle that logic separately on consumer end for now 🤔

Reply:
One special case is if a product wants to handle layout differently for landscape / portrait mode